homeassistant/config_entries.py

```diff
@@ -1359,8 +1429,15 @@ def async_update_entry(
         """
         changed = False
 
+        if unique_id is not UNDEFINED and entry.unique_id != unique_id:
+            # Reindex the entry if the unique_id has changed
```

Comment:
Maybe place as helper in the container class?

Reply:
I wanted to do something like than, but in the end I didn't do that because I'd have to mutate the entry object in the container class, and I wanted `async_update_entry` to be the only place that happens.